lib/logstash/inputs/http.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/namespace"
+require "stud/interval"
+require "socket" # for Socket.gethostname
+require "puma/server"
+require "puma/minissl"
+
+class Puma::Server
+  # ensure this method doesn't mess up our vanilla request
+  def normalize_env(env, client); end
+end
+
+# This output opens a web server and listens to HTTP requests,
+# converting them to LogStash::Event instances.
+#
+class LogStash::Inputs::Http < LogStash::Inputs::Base
+  config_name "http"
+
+  # If undefined, Logstash will complain, even if codec is unused.
+  default :codec, "plain"
+
+  # Which host or ip to bind to
+  config :host, :validate => :string, :default => "0.0.0.0"
+
+  # Which TCP port to bind to
+  config :port, :validate => :number, :default => 8080
+
+  # Maximum number of threads to use
+  config :threads, :validate => :number, :default => 4
+
+  # SSL Configurations
+  #
+  # Enable SSL
+  config :ssl, :validate => :boolean, :default => false
+
+  # The JKS keystore to validate the client's certificates
+  config :keystore, :validate => :path
+
+  #TODO: config :cacert, :validate => :path
+
+  # Set the truststore password
+  config :keystore_password, :validate => :password
```

Comment:
Maybe point to some doc to setup ssl? I've personally never used keystore jks before.